### PR TITLE
fixes #23846 - port robottelo docker repo tests

### PR DESCRIPTION
**Pull request by pondrejk**
Port robottelo tier1 api tests for docker repos
part of robottelo minitest port project: https://github.com/SatelliteQE/robottelo/projects/1

Related Robottelo issue: SatelliteQE/robottelo#6003

### CODE REVIEW

**Comment by jlsherrill** [comment]
[test katello]

**Comment by jturel** [comment]
@pondrejk Thanks for the added test coverage. We've been adding these similar PRs onto Katello 3.8 which is the release following the in-flight 3.7 since we did not see the need to immediately include them. Is that OK from your side? Just wanted to make you aware of our decision making.

File: test/models/repository_test.rb [hunk 1]
```diff
@@ -58,6 +59,19 @@ def test_docker_repository_docker_upstream_name_url
       assert @repo.valid?
     end
 
+    test_attributes :pid => '7967e6b5-c206-4ad0-bcf5-64a7ce85233b'
+    def test_docker_repository_update_name
+      @repo = build(:katello_repository, :docker,
+                    :environment => @library,
+                    :product => katello_products(:fedora),
+                    :content_view_version => @library.default_content_view_version
+                   )
```
Comment (ldjebran) [inline test/models/repository_test.rb]: The test say update but the @repo does not exist yet, wehave a missing @repo.save! here

Comment (jlsherrill) [reply test/models/repository_test.rb]: When testing validations, there is no need to save, so i think this is fine.

**Comment by jlsherrill** [comment]
Thanks @pondrejk !

**Pull request merged**